tests/integrations/crossed_integrations/test_sqs.py

```diff
@@ -0,0 +1,236 @@
+        )
+
+    @missing_feature(library="golang", reason="Expected to fail, Golang does not propagate context")
+    @missing_feature(library="ruby", reason="Expected to fail, Ruby does not propagate context")
+    @missing_feature(library="python", reason="Expected to fail, Python does not propagate context")
+    @missing_feature(library="nodejs", reason="Expected to fail, Nodejs does not propagate context")
+    @missing_feature(library="java", reason="Expected to fail, Nodejs does not propagate context")
+    def test_consume_trace_equality(self):
+        """This test relies on the setup for consume, it currently cannot be run on its own"""
+        producer_span = self.get_span(
+            self.buddy_interface,
+            span_kind=["producer", "client"],
+            queue=self.BUDDY_TO_WEBLOG_QUEUE,
+            operation="sendMessage",
+        )
+        consumer_span = self.get_span(
+            interfaces.library,
+            span_kind=["consumer", "client"],
+            queue=self.BUDDY_TO_WEBLOG_QUEUE,
+            operation="receiveMessage",
+        )
+
+        # Both producer and consumer spans should be part of the same trace
+        # Different tracers can handle the exact propagation differently, so for now, this test avoids
+        # asserting on direct parent/child relationships
+        assert producer_span["trace_id"] == consumer_span["trace_id"]
+
+    def validate_sqs_spans(self, producer_interface, consumer_interface, queue):
+        """
+        Validates production/consumption of sqs message.
+        It works the same for both test_produce and test_consume
+        """
+
+        # Check that the producer did not created any consumer span
+        assert (
+            self.get_span(producer_interface, span_kind=["consumer", "client"], queue=queue, operation="receiveMessage")
+            is None
+        )
+
+        # Check that the consumer did not created any producer span
+        assert (
+            self.get_span(consumer_interface, span_kind=["producer", "client"], queue=queue, operation="sendMessage")
+            is None
+        )
+
+        producer_span = self.get_span(
+            producer_interface, span_kind=["producer", "client"], queue=queue, operation="sendMessage"
+        )
+        consumer_span = self.get_span(
+            consumer_interface, span_kind=["consumer", "client"], queue=queue, operation="receiveMessage"
+        )
+        # check that both consumer and producer spans exists
+        assert producer_span is not None
+        assert consumer_span is not None
+
+        # Assert that the consumer span is not the root
+        assert "parent_id" in consumer_span, "parent_id is missing in consumer span"
+
+        # returns both span for any custom check
+        return producer_span, consumer_span
+
+
+@scenarios.crossed_tracing_libraries
+@coverage.basic
+@features.aws_sqs_span_creationcontext_propagation_with_dd_trace_js
+class Test_NodeJS_SQS(_Test_SQS):
```

Comment:
Having a test class per language is awkward, since we have to add one entry to every manifest for it, and it will also make them different rows in feature parity dashboards. I would suggst using a single `Test_SQS`, with per-language parameterization. See https://github.com/DataDog/system-tests/blob/3f93874f6c17f0d5ed43984f7a1adf349d44dfbe/tests/appsec/iast/source/test_parameter_value.py#L19-L20 for an example.

Reply:
Okay, I've re-worked the test classes. Now there are only two test classes, and they test one feature each.